### PR TITLE
[Merged by Bors] - chore(Order/Intervals/Set): move some defs to a new file

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Interval/Set/Basic.lean [hunk 1]
```diff
@@ -38,37 +38,6 @@ section Preorder
...
-
-/-- `Ici a` is the left-closed right-infinite interval $[a, ∞)$. -/
-def Ici (a : α) :=
-  { x | a ≤ x }
-
-/-- `Ioi a` is the left-open right-infinite interval $(a, ∞)$. -/
-def Ioi (a : α) :=
-  { x | a < x }
 
 theorem Ioo_def (a b : α) : { x | a < x ∧ x < b } = Ioo a b :=
```
Comment (grunweg) [inline Mathlib/Order/Interval/Set/Basic.lean:42]: Should these results also move into `Defs`?

File: Mathlib/Order/Interval/Set/Basic.lean [hunk 2]
```diff
@@ -38,37 +38,6 @@ section Preorder
 
 variable [Preorder α] {a a₁ a₂ b b₁ b₂ c x : α}
 
-/-- `Ioo a b` is the left-open right-open interval $(a, b)$. -/
```
Comment (grunweg) [inline Mathlib/Order/Interval/Set/Basic.lean:41]: The module docstring still talks about "contains these definitions" - this should be updated, right?

**Review by grunweg — APPROVED** [review]
Two small comments, otherwise LGTM. Thanks!

File: Mathlib/Order/Interval/Set/Basic.lean (hunk 2 above)
Comment (urkud) [reply Mathlib/Order/Interval/Set/Basic.lean:41]: Done.

File: Mathlib/Order/Interval/Set/Basic.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Order/Interval/Set/Basic.lean]: I have no preference here. Moved.

**Comment by grunweg** [comment]
Thanks!
maintainer merge

**Comment by grunweg** [comment]
(CI fails because of a linter warning: I just pushed a trivial fix.)

**Comment by YaelDillies** [comment]
Can you add some `assert_not_exists` to show the progress made on import reduction?

**Comment by urkud** [comment]
I'm not sure which of the newly available `assert_not_exists` I want to fix for the future. @YaelDillies Do you have preferences?

**Comment by YaelDillies** [comment]
What files are not imported by Data.Nat.Log anymore? I tried getting the info myself but failed. Within these, is there an obvious one that should not be imported in Data.Nat.Log ever again?

**Comment by urkud** [comment]
<details>
<summary>Mathlib files no longer imported by `Data.Nat.Log`</summary>
-      "name": "Mathlib.Data.Option.Defs",
-      "name": "Mathlib.Data.Set.Basic",
-      "name": "Mathlib.Data.Set.Operations",
-      "name": "Mathlib.Data.Set.Subsingleton",
-      "name": "Mathlib.Data.Sigma.Basic",
-      "name": "Mathlib.Data.SProd",
-      "name": "Mathlib.Data.Sum.Basic",
-      "name": "Mathlib.Lean.Elab.Tactic.Basic",
-      "name": "Mathlib.Lean.Meta.Basic",
-      "name": "Mathlib.Logic.Equiv.Basic",
-      "name": "Mathlib.Logic.OpClass",
-      "name": "Mathlib.Order.BooleanAlgebra",
-      "name": "Mathlib.Order.BoundedOrder",
-      "name": "Mathlib.Order.Disjoint",
-      "name": "Mathlib.Order.Heyting.Basic",
-      "name": "Mathlib.Order.Interval.Set.Basic",
-      "name": "Mathlib.Order.MinMax",
-      "name": "Mathlib.Order.PropInstances",
-      "name": "Mathlib.Order.SymmDiff",
-      "name": "Mathlib.Tactic.ByContra",
-      "name": "Mathlib.Tactic.CasesM",
-      "name": "Mathlib.Tactic.CC",
-      "name": "Mathlib.Tactic.CC.Addition",
-      "name": "Mathlib.Tactic.CC.Datatypes",
-      "name": "Mathlib.Tactic.CC.Lemmas",
-      "name": "Mathlib.Tactic.DeriveToExpr",
-      "name": "Mathlib.Tactic.FBinop",
-      "name": "Mathlib.Tactic.GeneralizeProofs",
-      "name": "Mathlib.Tactic.Lift",
-      "name": "Mathlib.Tactic.Relation.Rfl",
-      "name": "Mathlib.Tactic.Relation.Symm",
-      "name": "Mathlib.Tactic.Says",
-      "name": "Mathlib.Tactic.Tauto",
-      "name": "Mathlib.Tactic.ToExpr",
-      "name": "Mathlib.Tactic.ToLevel",
-      "name": "Mathlib.Util.Delaborators",
-      "name": "Mathlib.Util.WhatsNew",
</details>

**Comment by YaelDillies** [comment]
Nothing in this list jumps out as particularly advanced. Do we really want to do this split?

**Comment by YaelDillies** [comment]
The imports are already pretty minimal in the non-def file, so I don't really get the point, no.

**Comment by kim-em** [comment]
before
after

Hence adding `assert_not_exists OrderTop` in `Data.Nat.Log` is good here (and this opens the possibility of making `Nat.Log` even more basic. `Set.Basic` is too big, so this is good progress.

**Comment by kim-em** [comment]
bors merge

**Comment by YaelDillies** [comment]
> Set.Basic is too big

So `Set.Basic` is the file we should be splitting, not `Order.Intervals.Set`?

**Pull request closed without merging**